idb/src/main/kotlin/io/github/paulgriffith/kindling/idb/metrics/MetricTree.kt

```diff
@@ -0,0 +1,91 @@
+package io.github.paulgriffith.kindling.idb.metrics
+
+import com.jidesoft.swing.CheckBoxTree
+import io.github.paulgriffith.kindling.utils.AbstractTreeNode
+import io.github.paulgriffith.kindling.utils.TypedTreeNode
+import io.github.paulgriffith.kindling.utils.treeCellRenderer
+import javax.swing.tree.DefaultTreeModel
+import javax.swing.tree.TreePath
+
+data class MetricNode(override val userObject: String) : TypedTreeNode<String>()
+
+class MetricModel(metrics: List<Metric>) : DefaultTreeModel(RootNode(metrics))
+
+class RootNode(metrics: List<Metric>) : AbstractTreeNode() {
+    init {
+        val legacy = MetricNode("Legacy")
+        children.add(legacy)
+        val modern = MetricNode("New")
+        children.add(modern)
+
+        val seen = mutableMapOf<List<String>, MetricNode>()
+        for (metric in metrics) {
+            var lastSeen: MetricNode = if (metric.isLegacy) legacy else modern
+            val currentLeadingPath = mutableListOf(lastSeen.userObject)
+            for (part in metric.name.split('.')) {
+                currentLeadingPath.add(part)
+                val next = seen.getOrPut(currentLeadingPath.toList()) {
+                    val newChild = MetricNode(
+                        currentLeadingPath.drop(1).joinToString(separator = "."),
+                    )
+                    lastSeen.children.add(newChild)
+                    newChild
+                }
+                lastSeen = next
+            }
+        }
+    }
+
+    private val Metric.isLegacy: Boolean
+        get() = name.any { it.isUpperCase() }
+}
+
```

Comment:
Database names are in the db connection metric names, so the modern db metrics are under Legacy in the tree.